core/module.cc

```diff
@@ -474,7 +479,10 @@ int Module::ConnectModules(gate_idx_t ogate_idx, Module *m_next,
   igate->PushOgate(ogate);
 
   // Update graph
-  return ModuleBuilder::AddEdge(name(), m_next->name());
+  if (ModuleBuilder::AddEdge(name_, m_next->name_)) {
```

Comment:
You could just return !ModuleBuilder::AddEdge(...) if I am understanding the return value / logic right.